### PR TITLE
Added option to supply TestExecution result via REST endpoint

### PR DESCRIPTION
**Pull request by DreierF**
- [ ] [No new Teamscale findings](https://build.cqse.eu/teamscale/delta.html#input/jacoco-client/?showMergeFindings=true)
- [ ] [Changes are tested adequately](https://build.cqse.eu/teamscale/tests.html#/jacoco-client/)
- [ ] README.md updated in case of user-visible changes
- [ ] CHANGELOG.md updated
- [ ] build.gradle version number increased according to semantic versioning

### CODE REVIEW

**Comment by DreierF** [comment]
Fixes #68

File: agent/src/main/java/com/teamscale/jacoco/agent/AgentOptions.java [hunk 1]
```diff
@@ -199,21 +200,25 @@ public String createJacocoAgentOptions() {
...
-			return "sessionid=,destfile=" + new File(outputDirectory.toFile(),
-					"jacoco-" + dateFormat.format(new Date()) + ".exec").getAbsolutePath();
+			return "sessionid=,destfile=" + getTempFile("jacoco", "exec").getAbsolutePath();
 		} else {
 			return "output=none";
 		}
 	}
 
+	private File getTempFile(final String prefix, final String extension) {
+		DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss.SSS", Locale.US);
```
Comment (albertsteckermeier) [inline agent/src/main/java/com/teamscale/jacoco/agent/AgentOptions.java]: Why do we hard-code `Locale.US` here?

File: agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestExecutionWriter.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		}
+		FileReader fileReader = new FileReader(testExecutionFile);
+		List<TestExecution> testExecutions = GSON
+				.fromJson(fileReader, new TypeToken<List<TestExecution>>() {
+				}.getType());
+		fileReader.close();
+		testExecutions.add(testExecution);
+		FileWriter writer = new FileWriter(testExecutionFile);
+		GSON.toJson(testExecutions, writer);
+		writer.close();
```
Comment (albertsteckermeier) [inline agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestExecutionWriter.java]: Why do we read the whole file and then write it again for adding a test execution. Can't we write the result just once for all test executions?

Comment (DreierF) [reply agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestExecutionWriter.java]: The problem is that we will never know when we are done with the test execution.
We could only collect the test executions in memory and write them all to disk each time the list is updated.

File: agent/src/main/java/com/teamscale/jacoco/agent/AgentOptions.java (hunk 1 above)
Comment (DreierF) [reply agent/src/main/java/com/teamscale/jacoco/agent/AgentOptions.java]: Not sure why this ended up here. Removed it

**Pull request merged**